misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ActionNodeVisitor.java

```diff
@@ -154,11 +165,24 @@ public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
 
     @Override
     public void visit(FunctionCallExpressionNode functionCallExpressionNode) {
-
-        if (functionCallExpressionNode.functionName() instanceof SimpleNameReferenceNode) {
-            String methodName = ((SimpleNameReferenceNode) functionCallExpressionNode.functionName()).name().text();
+        // TODO: functionCallExpressionNode.functionName can be QualifiedNameReferenceNode as well
+        if ((functionCallExpressionNode.functionName() instanceof SimpleNameReferenceNode) &&
+                visitedActionNodes.stream().noneMatch(node -> {
+                    if (node.kind().equals(SyntaxKind.FUNCTION_CALL)) {
+                        NameReferenceNode funcName = ((FunctionCallExpressionNode) node).functionName();
+                        if (funcName.kind().equals(SyntaxKind.SIMPLE_NAME_REFERENCE)) {
+                            return ((SimpleNameReferenceNode) funcName).name().text()
+                                    .equals(((SimpleNameReferenceNode) functionCallExpressionNode.functionName())
+                                            .name().text());
+                        }
+                    }
+                    return false;
+                }
+                )) {
```

Comment:
I guess we can keep only the function names of the visited function nodes instead

Reply:
Instead of having function names as Strings, decided to have them as NameReferenceNodes, as for further processing having function names as Strings does not allow it.

Reply:
Yeah it makes sense. +1